core/node/api_server/src/execution_sandbox/mod.rs

```diff
@@ -417,3 +417,28 @@ impl BlockArgs {
         )
     }
 }
+
+/// VM executor capable of executing isolated transactions / calls (as opposed to batch execution).
+#[async_trait]
+trait OneshotExecutor<S: ReadStorage> {
+    type Tracers: Default;
```

Comment:
I don't like this associated type very much; probably would need to get rid of it in the future refactoring iterations (see the overall comment) in favor of something like

```rust
pub struct Tracers<'a> {
    pub calls: Option<&'a mut Vec<Call>>,
    pub validation: Option<(
        ValidationTracerParams,
        &'a mut Result<(), ViolatedValidationRule>,
    )>,
}
```